### ISSUE TITLE
unresponsive interface issues, i believe due to (potentially?) unnecessary recursion

### ISSUE THREAD

**Issue by AlexRyanUSACE** [HydrologicEngineeringCenter/HEC-FDA#1006]
There are some cases where the interface is unresponsive, and it appears the reason is a lot of recursive code where user changes result in a pretty large tree of iterative code execution, some of which looks like it is either unneeded or could be substantially optimized. As an example, when editing data table values as in the Create Frequency Function interface, changing a single cell value results in a pretty sizeable amount of iterative logic that ends up taking around 15 seconds on my machine, during which the program is unresponsive. If only 8 values in tabular form are pasted into the table, the program is unresponsive for 5+ minutes (i didn't wait for it to finish). Mapping out just this code execution looks (roughly) like the following:

![image](https://github.com/HydrologicEngineeringCenter/HEC-FDA/assets/7873755/e14f9b2f-faad-467d-863e-58e4deef2187)

just to, from what i can tell at least, reset all of the confidence values to 0. There are a number of unnecessary execution branches (i noted some of these in red) that can be seen in this but i suspect the rest could be optimized as well. 

This is really only a usability issue/annoyance when entering tabular data cell at a time, but the interface should support copy/pasting tabular data and if it takes upwards of 5 minutes to paste in only 8 values that functionality is effectively not there.

**Comment by Brennan1994** [comment]
Yeah, this is a nightmare of a chunk of code. It was a control i stole whole sale from a side project I was working on to learn WPF.  It could use more love, but There's a fix coming in the next PR that will address the most egregious code.
